core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -153,6 +158,10 @@ object Semaphore {
 
           ref.modify(loop(n, _, ZIO.unit)).flatten
         }
+        def awaiting(implicit trace: Trace): UIO[Long] = ref.get.map {
+          case Left(queue) => queue.size
+          case Right(_)    => 0L
+        }
```

Comment:
You can avoid a flatMap by using getWith

```suggestion
        def awaiting(implicit trace: Trace): UIO[Long] = ref.get.getWith {
          case Left(queue) => Exit.succeed(queue.size)
          case Right(_)    => Exit.succeed(0L)
        }
```